src/base/bittorrent/session.cpp

```diff
@@ -2367,7 +2368,7 @@ void Session::exportTorrentFile(TorrentHandle *const torrent, TorrentExportFolde
 
 void Session::generateResumeData(bool final)
 {
-    foreach (TorrentHandle *const torrent, m_torrents) {
+    for (TorrentHandle *const torrent : torrents()) {
         if (!torrent->isValid()) continue;
```

Comment:
Probably
```c++
for (const TorrentHandle * const torrent : copyAsConst(torrents())
```
is more correct.

Reply:
It's unattended change so it will be reverted.